frontend/packages/ceph-storage-plugin/src/components/block-pool/body.tsx

```diff
@@ -0,0 +1,258 @@
+
+import './body.scss';
+
+export const BlockPoolStatus: React.FC<BlockPoolStatusProps> = ({ status, name, error = '' }) => {
+  const { t } = useTranslation();
+  const statusObj = PROGRESS_STATUS(t).find((state) => state.name === status);
+
+  return (
+    <>
+      <EmptyState>
+        <EmptyStateIcon icon={statusObj.icon} className={statusObj.className} />
+        <EmptyStateBody data-test="empty-state-body">
+          {error ? getErrorMessage(error) : statusObj.desc.replace('{name}', name)}
+        </EmptyStateBody>
+      </EmptyState>
+    </>
+  );
+};
+
+export type BlockPoolStatusProps = {
+  status: string;
+  name?: string;
+  error?: string;
+};
+
+export const BlockPoolBody = (props: BlockPoolBodyPros) => {
+  const { cephCluster, state, dispatch } = props;
+  const { t } = useTranslation();
+
+  const isPoolManagementSupported = useFlag(GUARDED_FEATURES.OCS_POOL_MANAGEMENT);
+  const [storageCluster, storageClusterLoaded, storageClusterLoadError] = useK8sGet<
+    ListKind<StorageClusterKind>
+  >(OCSServiceModel, null, CEPH_STORAGE_NAMESPACE);
+
+  const [isReplicaOpen, setReplicaOpen] = React.useState(false);
+  const [isVolumeTypeOpen, setVolumeTypeOpen] = React.useState(false);
+
+  // Check storage cluster is arbiter
+  React.useEffect(() => {
+    const isArbiterCluster: boolean = checkArbiterCluster(storageCluster?.items[0]);
+    dispatch({ type: BlockPoolActionType.SET_POOL_ARBITER, payload: isArbiterCluster });
+    if (isArbiterCluster) {
+      dispatch({ type: BlockPoolActionType.SET_POOL_REPLICA_SIZE, payload: '4' });
+    }
+  }, [
+    storageCluster,
+    storageClusterLoaded,
+    storageClusterLoadError,
+    state.isArbiterCluster,
+    dispatch,
+  ]);
+
+  const replicaList: string[] = _.keys(OCS_DEVICE_REPLICA).filter(
+    (replica: string) =>
+      (state.isArbiterCluster && replica === '4') || (!state.isArbiterCluster && replica !== '4'),
+  );
+
+  const replicaDropdownItems = replicaList.map((replica) => (
+    <DropdownItem
+      key={`replica-${OCS_DEVICE_REPLICA[replica]}`}
+      component="button"
+      id={replica}
+      data-test-id={replica}
+      onClick={(e) =>
+        dispatch({ type: BlockPoolActionType.SET_POOL_REPLICA_SIZE, payload: e.currentTarget.id })
+      }
+    >
+      {t('ceph-storage-plugin~{{replica}} Replication', { replica: OCS_DEVICE_REPLICA[replica] })}
+    </DropdownItem>
+  ));
+
+  // Volume Type
+  let availableDeviceClasses = [];
+  const deviceClasses = cephCluster?.status?.storage?.deviceClasses ?? [];
+  const setVolumeType = (volumeType: string) =>
+    dispatch({ type: BlockPoolActionType.SET_POOL_VOLUME_TYPE, payload: volumeType });
+
+  if (deviceClasses.length && isPoolManagementSupported) {
+    if (state.volumeType === '') {
+      // Set default value
+      const ssdDeviceClass = deviceClasses.find((deviceClass) => deviceClass.name === 'ssd') || {};
+      Object.keys(ssdDeviceClass).length
+        ? setVolumeType('ssd')
```

Comment:
@afreen23 instead of deviceClass.name i used 'ssd' here (one of your review comment is to use deviceClass.name)

since I am using default {} in L112 it is not allowing me to use deviceClass.name